priv/tools/internal/riak_cs_inspector.erl

```diff
@@ -259,41 +261,110 @@ cs_bucket_info(RiakBucket, KnownNames) ->
             ['riak-cs', RiakBucket, RiakBucket]
     end.
 
--spec list_cs_buckets(pid()) -> any().
-list_cs_buckets(RiakcPid)->
+list_cs_buckets(RiakcPid, ListFun)->
     heading("~-64..=s ~-8..=s: ~-40..=s~n",
             ["CS Bucket Name ", "Sibl. ", "Owner Key "]),
-    {ok, Keys} = riakc_pb_socket:list_keys(RiakcPid, "moss.buckets"),
-    lists:foreach(fun(Key) ->
-                          [io:format("~-64s ~-8B: ~-40s~n", [Key, SiblingNo, V])
-                           || {SiblingNo, {_MD, V}}
-                                  <- get_riak_object(RiakcPid, "moss.buckets", Key)]
-                  end, lists:sort(Keys)).
+    PrintFun = fun(Keys) ->
+                       lists:foreach(fun(Key) ->
+                                             [io:format("~-64s ~-8B: ~-40s~n", [Key, SiblingNo, V])
+                                              || {SiblingNo, {_MD, V}}
+                                                     <- get_riak_object(RiakcPid, "moss.buckets", Key)]
+                                     end, Keys)
+               end,
+    ListFun(RiakcPid, "moss.buckets", PrintFun).
+
+lookup_listbucket_fun(Opts) ->
+    case proplists:get_value(sort, Opts) of
+        true ->
+            fun pb_list_buckets/2;
+        _ ->
+            fun pb_stream_list_buckets/2
+    end.
 
+lookup_listkey_fun(Opts) ->
+    case proplists:get_value(sort, Opts) of
+        true ->
+            fun pb_list_keys/3;
+        _ ->
+            fun pb_stream_list_keys/3
+    end.
+
+% For Backward compatibility. Is this needed?
 list_objects(RiakcPid, Bucket)->
+    list_objects(RiakcPid, Bucket, fun pb_list_keys/3).
+
+list_objects(RiakcPid, Bucket, ListFun)->
     print_manifest_summary(pid, bucket, sibling_no, header),
     ManifestBucketName = riak_cs_utils:to_bucket_name(objects, Bucket),
-    {ok, ManifestKeys} = riakc_pb_socket:list_keys(RiakcPid, ManifestBucketName),
-    [print_manifest_summary(RiakcPid, Bucket, SiblingNo, M)
-     || Key <- lists:sort(ManifestKeys),
-        {SiblingNo, _UUID, M} <- get_manifest(RiakcPid, Bucket, Key)].
+    PrintFun = fun(ManifestKeys) ->
+                       [print_manifest_summary(RiakcPid, Bucket, SiblingNo, M)
+                        || Key <- ManifestKeys,
+                           {SiblingNo, _UUID, M} <- get_manifest(RiakcPid, Bucket, Key)]
+               end,
+    ListFun(RiakcPid, ManifestBucketName, PrintFun).
+
+pb_list_buckets(RiakcPid, PrintFun) ->
```

Comment:
Also, from this function to `print_stream_list()` can be moved to `%% Riak Utilities` part.